spring-cloud-gcp-pubsub/src/test/java/org/springframework/cloud/gcp/pubsub/support/converter/JacksonPubSubMessageConverterTests.java

```diff
@@ -66,6 +67,12 @@ public void testPojo() throws JSONException {
 		assertThat(o).as("verify that deserialized object is equal to the original one").isEqualTo(contact);
 	}
 
+	@Test
+	public void testToPubSubMessageWithNullPayload() throws JSONException {
+		PubsubMessage pubsubMessage = this.converter.toPubSubMessage(null, null);
```

Comment:
Is it really OK to publish a null body?
Then it really confirms my concern to move null check to the converter implementation instead of the `PubSubTemplate`.
Otherwise it collides with what you have done there and is testing here.

Reply:
`JacksonPubSubMessageConverter` and `SimplePubSubMessageConverter` both behave differently.

For `JacksonPubSubMessageConverter` a `null` payload results in a `PubSubMessage` with an empty `ByteString`.

For `SimplePubSubMessageConverter`, a `null` payload results in a `NullPiointerException`.

I would prefer to force the payload to not be null via an `Assert.notNull()`

Thoughts on the behavior change?

Reply:
I think we can leave it up to each converter to decide if `null` payload is OK.